_posts/andy_yang/2016-11-18-azure-java-examples.md

```diff
@@ -0,0 +1,228 @@
+---
+layout: post
+title:  "Azure Blob Storage - Java Examples"
+breadcrumb: true
+author: andy_yang
+date: 2016-11-18
+categories: andy_yang
+tags:
+    - Azure
+    - Java
+    - Cloud
+    - Microsoft
+teaser:
+    info: Examples in Java for working with Azure Blob Storage
+    image: andy_yang/azure/azure-logo.jpg
+header: 
+    version: small
+    title: Software Engineering Blog
+    image: header-logo-crop.png
+    icon: icon-blog
+---
+
+## Introduction
+The ICGC In The Cloud initiative makes a large genomic database available to cancer researchers in cloud computing environments. Cloud computing helps alleviate the burden, in transfer time and computing resources, that the size of the ICGC datasets places on research groups. Currently, over 750 TB of genomic data is hosted between Amazon Web Services and an OpenStack environment called Cancer Collaboratory. Recently, Microsoft has also expressed a willingness to join ICGC as a partner in hosting genomic data. 
+
+Both OpenStack's Ceph ObjectStore and AWS S3 use the S3 API. With a very small difference, we are able to essentially run a single code base against both repositories. Microsoft's Azure Blob Storage is altogether a different beast however, and I recently became familiar with it in order to adapt our Storage system to use Azure as a backing repository.
+
+
+## Background
+
+### Architecture
+
+<center>
+  <figure style="width: 85%;">
+      <img src="{{site.urlimg}}/andy_yang/azure/storage-topology.png"/>
+      <figcaption>Storage System Topology</figcaption>
+  </figure>
+</center>
+
+At a high level, our system is comprised of 3 components. In addition to the repository/object store such as AWS S3, Ceph or Azure Blob Storage, we have a Storage Server component, as well as a Storage Client.
+
+We introduce a middle tier Storage Server component that works in concert with the Client in order to restrict access to the data. Our genomic data is private and subject to tight control. The underlying repository is locked-down (i.e., no anonymous/direct outside access).
+
+The Storage Server component is primarily concerned with determining permissions and generating upload/download URL's. All actual transfers from the repository are initiated and managed directly from the client. The Storage Server is no longer involved by this stage.
```

Comment:
It is no longer involved because the client is connecting to the repository directly - the server is not in-between or anything.